core/services/fluxmonitor/flux_monitor.go

```diff
@@ -250,7 +252,7 @@ func (f pollingDeviationCheckerFactory) New(
 	}
 
 	fetcher, err := newMedianFetcherFromURLs(
-		timeout,
+		timeout.Duration(),
```

Comment:
Shouldn't the `models.Duration` be pushed further down here?

Reply:
That's a nice-to-have, but this is a call to an internal method, and we should really only have to protect `fluxmonitor` at its interfaces.

Reply:
Does "That's a nice to have" just mean "that's out of scope"?

As a rule, I don't entirely agree with `and we should really only have to protect fluxmonitor at its interfaces`, given your PR doesn't really do that - it introduces a new type! And ... flux monitor's interface still accepts negative durations.. maybe some confusion in terminology here.

This may be an appropriate compromise, but there are definitely some ways negative values could sneak through after this PR.

Reply:
Thanks, you're right that `NewPollingDeviationChecker` is logically part of fluxmonitor's external interface (I ignored it because we don't actually use it externally), and also that it's much better to just use `models.Duration` everywhere. I've removed all instances of `time.Duration` from `core/services/fluxmonitor`. Can I get a re-approval?